### PR TITLE
Upgrade to rails 4.2.7.1

### PR DESCRIPTION
**Pull request by h-lame**
This is the minimal update set to drag us from 4.1 to 4.2.  We need to follow up with some other maintenance on dependencies as many are out of date.

See notes in individual commits for why each change (other than Gemfile changes) were made.

### CODE REVIEW

File: db/schema.rb [hunk 1]
```diff
@@ -98,92 +98,92 @@
...
     t.datetime "created_at"
     t.datetime "updated_at"
-    t.text     "street_address"
-    t.string   "locality"
-    t.string   "region"
-    t.string   "postal_code"
-    t.string   "world_location_slug"
-    t.integer  "contact_type_id"
-    t.decimal  "latitude",            precision: 10, scale: 0
-    t.decimal  "longitude",           precision: 10, scale: 0
```
Comment (brenetic) [inline db/schema.rb]: Do we need `scale` here?

File: db/schema.rb [hunk 2]
```diff
@@ -98,92 +98,92 @@
...
+    t.text     "street_address",      limit: 65535
+    t.string   "locality",            limit: 255
+    t.string   "region",              limit: 255
+    t.string   "postal_code",         limit: 255
+    t.string   "world_location_slug", limit: 255
+    t.integer  "contact_type_id",     limit: 4
+    t.decimal  "latitude",                          precision: 10
+    t.decimal  "longitude",                         precision: 10
```
Comment (brenetic) [inline db/schema.rb]: Do we need `scale` here?

**Comment by brenetic** [comment]
@h-lame DB schema change looks like it's not adding anything extra. Just the one question on that. Otherwise all 🍑

File: db/schema.rb (hunk 2 above)
Comment (h-lame) [inline db/schema.rb]: A good question.  I'm guessing that scale: 0 is probably the default, but I'll check.  Note that if we do need it I'm not sure entirely how we'd generate it.

Comment (h-lame) [inline db/schema.rb]: Actually, I think I'm going to drop this commit.  I don't think we should have these limits here at all, and in some cases they don't seem legit either.  `contact_id` on this table is an int(11) in my db, but it's setting `limit: 4` here.

**Comment by brenetic** [comment]
👍

**Pull request merged**